common/src/test/scala/org/neo4j/spark/util/ValidationsIT.scala

```diff
@@ -0,0 +1,180 @@
+package org.neo4j.spark.util
+
+import org.junit.rules.ExpectedException
+import org.junit.{Rule, Test}
+import org.neo4j.driver.AccessMode
+import org.neo4j.spark.{SparkConnectorScalaSuiteIT, TestUtil}
+
+class ValidationsIT extends SparkConnectorScalaSuiteIT {
+
+  val _expectedException: ExpectedException = ExpectedException.none
+
+  @Rule
+  def exceptionRule: ExpectedException = _expectedException
+
+  @Test
+  def testReadQueryShouldBeSyntacticallyInvalid(): Unit = {
+    // then
+    val expectedMessage = if (TestUtil.neo4jVersion().startsWith("4")) {
```

Comment:
this assertion may be a bit too strict and may unfortunately change again in future versions, which will require updates to this test
maybe just check for "Invalid input"?